New location picker release adding new configuration options and integrates with new ACPaaS api

Florian-Ramakers opened the pull request:
- New Location Picker major release v4.0.0
- Integrates with its own ACPaaS API

Live demo on: https://locationpicker-app1-o.antwerpen.be/

jsebrech commented:
Why did you remove changelog.md, contributing.md and license?

jsebrech commented on README.md:
```diff
@@ -1,112 +1,141 @@
+    /* the zoom level when a location is selected. */
+    onSelectZoom: number = 16;
+    /* the initial map center on load. */
+    mapCenter: Array<number> = [51.215, 4.425];
```
does this replace the location property for setting an initial location? what if the application just has an address and wants to show this field prefilled? how do you use this in a form which was already filled out and address is one of the filled out values?

Florian-Ramakers replied:
Initial location can now be set using [(NgModel)] or a reactive form control

jsebrech commented on projects/ngx-location-picker/src/lib/components/ngx-location-picker.component.html:
```diff
@@ -0,0 +1,53 @@
+                        (click)="emptyField()"
+                        [disabled]="inputHasValue"
+                >
+                    <span class="show-on-desktop">Leegmaken</span>
```
make this an icon and use an aria label instead (but the text of that should be configurable, just like all localizable text)

jsebrech commented on projects/ngx-location-picker/src/lib/components/ngx-location-picker.component.spec.ts:
```diff
@@ -0,0 +1,39 @@
+    it('should create', () => {
+        expect(component).toBeTruthy();
+    });
+});
```
the previous tests should be migrated to the new version

Florian-Ramakers replied:
Added new tests

jsebrech commented on projects/ngx-location-picker/src/lib/components/ngx-location-picker.component.ts:
```diff
@@ -0,0 +1,391 @@
+    @Output() editFeature = new EventEmitter<any>();
+    /* locationSelect event: fired when selecting a location. */
+    @Output() locationSelect = new EventEmitter<LocationModel | AddressModel | CoordinateModel>();
+
```
the old control supported value and valueChange properties. This allows two-way binding the control: `<ngx-location-picker [(value)]="locationValue">

It also implemented the ControlValueAccessor interface so the control could be used in the ngModel style: 

- https://angular.io/api/forms/NgModel
- https://angular.io/guide/forms#two-way-data-binding-with-ngmodel

I think it is an important requirement that you can use it in a form easily as a location field.

jsebrech commented on projects/ngx-location-picker/src/lib/components/ngx-location-picker.component.html:
```diff
@@ -0,0 +1,53 @@
+            </div>
+        </div>
+
+        <div class="m-flyout m-flyout--full" [class.is-open]="didSearch">
```
is there a specific reason for not using the autocomplete control anymore?

Florian-Ramakers replied:
Had a few issues using the autocomplete in a previous project (styling related) decided to just use an input field and the flyout html from the core branding github repo.

Florian-Ramakers commented on README.md:
```diff
@@ -1,112 +1,141 @@
+# NgxLocationPicker
 
-This is the Angular 6+ UI for a Smart Widget implementing a picker field to choose a location (street, address or point of interest). It is matched by a [corresponding back-end service](https://github.com/digipolisantwerp/location-picker_service_nodejs) which is needed when running it in remote mode. A default implementation for selecting locations in antwerp is provided.
+Location Picker for Angular 7+. Provides easy to use interface for searching locations or addresses within the city of Antwerp.
```
This library was generated using Angular 7 (and should work with 8 as well). I didn't have the chance yet to test it in Angular 6.

jsebrech replied:
Please make it v6 compatible if possible.

Florian-Ramakers replied:
Component was tested in 6, 7 and 8. Updated readme.

Florian-Ramakers commented on README.md:
````diff
@@ -1,112 +1,141 @@
 
 ```ts
-import { LocationPickerModule } from '@acpaas-ui-widgets/ngx-location-picker';
+import {NgxLocationPickerModule} from '@acpaas-ui-widgets/ngx-location-picker';
````
This library was originally created in context of another project that integrates with the new location picker api. We later on decided that this library could be useful for other Digipolis projects as well and that this could be a new release of the existing location picker, hence the large amount of changes. I'll review all comments and provide updates to make sure everything is in line with this repo.

Florian-Ramakers commented on README.md:
````diff
@@ -1,112 +1,141 @@
 ```
 
-### In your template:
+##### Getting an API key
````
I don't think this repo is the right place to explain the setup of a BFF. Ideally the API Key explanation is moved to the original location picker BFF repo (like you suggested). A new major release of the old Location Picker BFF would be needed where we reduce the old BFF to a simple proxy including the API Key. Apart from Node.JS we should probably include a .NET core based BFF example as well.

I'll remove the BFF and API Key part from the readme.

Florian-Ramakers commented on angular.json:
```diff
@@ -98,41 +139,73 @@
-    "location-picker_widget_angular-e2e": {
-      "root": "e2e",
-      "sourceRoot": "e2e",
+    "example-e2e": {
```
There is one default test included but it doesn't do much. Better to remove it and stick to unit tests.

Florian-Ramakers commented on README.md:
````diff
@@ -1,112 +1,141 @@
 <link rel="stylesheet" href="https://cdn.antwerpen.be/core_branding_scss/3.0.3/main.min.css">
 ```
 
-> For projects that are still using Angular 5, we are [maintaining a v1 branch](https://github.com/digipolisantwerp/location-picker_widget_angular/tree/v1), which will still receive bug fixes if needed.
````
I've updated the readme to include this info.

Florian-Ramakers commented on README.md:
```diff
@@ -1,112 +1,141 @@
+    /* show or hide the map. */
+    showMap: boolean = true;
+    /* add layers to show on the map. eg: A-card terminals, Velo stations, ... */
+    featureLayers: FeatureLayerModel[] = [];
```
now explained in readme and in interface file

jsebrech commented on projects/ngx-location-picker/src/lib/components/ngx-location-picker.component.css:
```diff
@@ -1,10 +1,17 @@
+    font-size: 40px;
+}
+
+::ng-deep .is-picking .leaflet-grab {
```
core-branding-gewijs werken we normaal gezien met de font awesome icons, bvb deze: https://fontawesome.com/icons/map-pin?style=solid

Florian-Ramakers replied:
added suggested marker

Florian-Ramakers commented:
Ik heb de pull request geüpdatet om naar de V4 branch te gaan om eerst wat meer te testen in combinatie met de nieuwe location picker service.

TriangleJuice commented on package.json:
```diff
@@ -1,76 +1,59 @@
-    "@angular/platform-browser": "6.1.9",
-    "@angular/platform-browser-dynamic": "6.1.9",
-    "@angular/router": "6.1.9",
+    "@a-ui/core": "^3.2.2",
```
```suggestion
    "@a-ui/core": "^4.0.0",
```

TriangleJuice commented on projects/example/src/index.html:
```diff
@@ -0,0 +1,28 @@
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <link rel="icon" type="image/x-icon" href="favicon.ico">
+
+    <link rel="stylesheet" href="https://cdn.antwerpen.be/core_branding_scss/3.2.2/main.min.css">
```
```suggestion
    <link rel="stylesheet" href="https://cdn.antwerpen.be/core_branding_scss/4.0.0/main.min.css">
```

TriangleJuice commented on projects/ngx-location-picker/src/lib/components/ngx-location-picker.component.html:
```diff
@@ -0,0 +1,110 @@
+          (click)="toggleTileLayer()"
+          [disabled]="isDefaultTileLayer"
+        >
+          Kaart
```
Dit label zou ook bij de props moeten gestoken worden, zodat het in meerdere talen kan aangeboden worden.

TriangleJuice commented on projects/ngx-location-picker/src/lib/components/ngx-location-picker.component.html:
```diff
@@ -0,0 +1,110 @@
+        (click)="pickLocationOnMap()"
+        [class.is-active]="pickLocationActive"
+      >
+        <i class="fa fa-map-marker" aria-hidden="true" aria-label="Pick location on map"></i>
```
Dit label zou ook bij de props moeten gestoken worden, zodat het in meerdere talen kan aangeboden worden.

TriangleJuice commented on projects/ngx-location-picker/src/lib/components/ngx-location-picker.component.html:
```diff
@@ -0,0 +1,110 @@
+          class="a-button a-button--small has-icon a-button--block button-zoom-in"
+          (click)="zoomIn()"
+        >
+          <i class="fa fa-plus" aria-hidden="true" aria-label="Zoom in"></i>
```
Dit label zou ook bij de props moeten gestoken worden, zodat het in meerdere talen kan aangeboden worden.

Florian-Ramakers commented on README.md:
```diff
@@ -32,81 +30,295 @@ import { LocationPickerModule } from '@acpaas-ui-widgets/ngx-location-picker';
+        /* The input field placeholder text. */
+        @Input() placeholder = 'Locaties zoeken...';
+        /* Label to show above the search field. */
+        @Input() label = '';
```
Dat label was overbodig met de nieuwe layout omdat de zoekbalk nu op de map staat en niet meer erboven.

jsebrech commented:
Bugs:

- Type "generaal armstrongweg 1", jump to that address (map pans: ok), type "markgravelei", select that street. The map pans but to neither markgravelei or generaal armstrongweg.
- Pan the map to outside antwerpen. Type "stadspark", select the park. Nothing appears to happen. Expected behavior: the map pans to the park.

jsebrech commented on projects/ngx-location-picker/src/lib/components/ngx-location-picker.component.ts:
```diff
@@ -0,0 +1,634 @@
+  /**
+   * Writes a new value to the element.
+   *
+   * @since 4.0.0
```
all these `@since` comments are not necessary imho, they just take up space

jsebrech commented on projects/ngx-location-picker/src/lib/components/ngx-location-picker.component.ts:
```diff
@@ -0,0 +1,634 @@
+     * Show notification if only scroll is used.
+     */
+    if (event.type === 'wheel' && this.cursorOnLeaflet) {
+      if (event.ctrlKey || event.metaKey) {
```
This conflicts in my browser (chrome on linux) with the browser's builtin ctrl-+ zoom feature. Maybe the shift key would be a better option?

Florian-Ramakers commented on package.json:
```diff
@@ -1,76 +1,59 @@
-    "@angular/core": "^6.0.0",
-    "@angular/forms": "^6.0.0",
-    "@acpaas-ui/ngx-components": "^1.0.3"
+  "dependencies": {
```
Yes!

Florian-Ramakers commented:
I have implemented the requested changes and fixed a few bugs. A beta version of this component is currently being tested in another project. Any bugs will be logged in JIRA and fixed in future versions eg 4.0.1, ...

Florian-Ramakers commented on README.md:
```diff
@@ -32,81 +30,303 @@ import { LocationPickerModule } from '@acpaas-ui-widgets/ngx-location-picker';
+
+Note: There are 3 methods of getting values after selecting a location:
+
+**Method 1: locationSelect event**  
```
locationChange now fires when input is cleared
